### PR TITLE
added enwiki and corpus sampler

### PR DESCRIPTION
**Pull request by jeremyasapp**
- Fixes a small issue with caching in TabularDataset
- Introduced the CorpusSampler for going through a corpus of data
- Introduced the Enwiki8 dataset

### CODE REVIEW

File: flambe/nlp/language_modeling/datasets.py [hunk 1]
```diff
@@ -96,14 +91,75 @@ def _process(self, file: bytes) -> List[Tuple[str]]:
...
+
+
+class Enwiki8(TabularDataset):
+    """The official WikiText103 dataset."""
+
+    ENWIKI_URL = "http://mattmahoney.net/dc/enwik8.zip"
+
+    def __init__(self,
+                 unroll_size: Optional[int] = 100,
+                 num_test_symbols: int = 5000000,
```
Comment (iitzco-asapp) [inline flambe/nlp/language_modeling/datasets.py]: What does `5000000` stand for? Forgot to add documentation for a couple of parameters (including `num_test_symbols` )

File: flambe/nlp/language_modeling/sampler.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import torch
+from torch.utils.data import DataLoader
+
+from flambe.sampler.sampler import Sampler
+
+
+class CorpusSampler(Sampler):
+    """Implement a CorpusSampler object.
+
+    This object is useful in ietrating over a large corpus of
```
Comment (iitzco-asapp) [inline flambe/nlp/language_modeling/sampler.py]: typo `iterating`

File: flambe/nlp/language_modeling/sampler.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
+from typing import Optional, Sequence, Tuple, Iterator
+import math
+
+import torch
+from torch.utils.data import DataLoader
+
+from flambe.sampler.sampler import Sampler
+
+
+class CorpusSampler(Sampler):
```
Comment (iitzco-asapp) [inline flambe/nlp/language_modeling/sampler.py]: Maybe it's worth inheriting from `BaseSampler`? I see at least in the `__init__` a lot of shared parameters.

File: flambe/nlp/language_modeling/sampler.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                 unroll_size: int = 100,
+                 n_workers: int = 0,
+                 pin_memory: bool = False,
+                 seed: Optional[int] = None,
+                 downsample: Optional[float] = None,
+                 drop_last: bool = True) -> None:
+        """Initialize the BaseSampler object.
+
+        Parameters
+        ----------
```
Comment (iitzco-asapp) [inline flambe/nlp/language_modeling/sampler.py]: Missing `unroll_size` in the docs

File: flambe/nlp/language_modeling/datasets.py (hunk 1 above)
Comment (jeremyasapp) [reply flambe/nlp/language_modeling/datasets.py]: Fixed

File: flambe/nlp/language_modeling/sampler.py (hunk 3 above)
Comment (jeremyasapp) [reply flambe/nlp/language_modeling/sampler.py]: Meh barely saves anything. Will go over this during refactor.

**Pull request merged**